Building.md

```diff
@@ -65,10 +65,17 @@ correct path to your cuQuantum installation by setting the environment variable
 
 ## Building CUDA Quantum with a custom LLVM version
 
-CUDA Quantum is intended to be built using the LLVM commit that the submodule is
-set to. If you need to update the LLVM commit, or if you are not using the
-pre-built version of that commit in our development container, LLVM will need to
-be built from source. To configure our build scripts, please set the LLVM
+CUDA Quantum is intended to be built using the LLVM commit that the submodule
+is set to (currently LLVM 16.0.6). If you need to update the LLVM commit, or if
+you are not using the pre-built version of that commit in our development
+container, LLVM will need to be built from source. ***Building with LLVM >=
+17.x is not currently supported*** due to the QIR specification's requirement
```

Comment:
We might want to just have a link to the issue number here rather than explain the QIR limitations and such in prose in the docs?

Reply:
@schweitzpgi - I think you're talking about Issue #16. (Let me know if you disagree.) @bettinaheim - let me know when you'd prefer. I know you had previously approved this PR before Eric's comment, but if you'd rather I link to Issue #16 per Eric's suggestion, just let me know.

Reply:
I think it is one sentence here that very clearly states that building with 17.x is not supported. The GitHub issue can get pretty technical, so I'd just keep it at that here.